### PR TITLE
[WE-2603] New VC display

### PR DESCRIPTION
**Pull request by smurflo2**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=vFbxbEh)
# Overview

#### 🎟 Relevant Jira Issues

[WE-2603](https://welibrary.atlassian.net/browse/WE-2603)

#### 📚 What is the context and goal of this PR?
We've got a new design for VC's! [Figma](https://www.figma.com/file/RCbnVvxZtrgWTIkelNpepG/Learn-Card-Wallet?node-id=3137%3A21101&t=5pOqTHEUYk2GLJgI-0)

This also has code for handling skills and a notes field, but those things are commented out right now since we're not using them

#### 💡 Feature Breakdown (screenshots & videos encouraged!)
With defaults:
image
image

Attachments:
image
(When an attachment is clicked it will be handled by a custom function passed in by the consumer)

Credential Verifications:
image

You can also customize many of the elements:
image
- The X appears if you specify a `onXClick` function (expected value is something like `closeModal`)
- The background can be set to a custom image or color
- The issuer and issuee image and name can be overridden
- The bottom right icon and background color can be customized

#### 🔍 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (refactor, documentation update, etc)

#### 💳 Does This Create Any New Technical Debt? ( If yes, please describe and [add JIRA TODOs](https://welibrary.atlassian.net/jira/software/projects/WE/boards/2) )
- [x] No
- [ ] Yes

# Testing

#### 🔬 How Can Someone QA This?

 - Start storybook and check out the stories for VCDisplayCard2 and VCCard
 - That should really cover it, but feel free to
   - mess with the props in storybook or modify the credentials in `test.helpers.ts`
   - import this component and use it in a different context

#### 🧪 Code Coverage

Storybook

# Documentation

#### 📊 Storybook

# ✅ PR Checklist
- [x] Related to a Jira issue ([create one if not](https://welibrary.atlassian.net/jira/software/projects/WE/boards/2))
- [x] My code follows **style guidelines** (eslint / prettier)
- [x] I have **manually tested** common end-2-end cases
- [x] I have **reviewed** my code
- [x] I have **commented** my code, particularly where ambiguous
- [x] New and existing **unit tests pass** locally with my changes
- [ ] I have made corresponding changes to **gitbook documentation**

### 🚀 Ready to squash-and-merge?:
- [x] Code is backwards compatible
- [x] There is **not** a "Do Not Merge" label on this PR
- [x] I have thoughtfully considered the security implications of this change.
- [x] This change does not expose new public facing endpoints that do not have authentication

### CODE REVIEW

**Comment by smurflo2** [comment]
@gerardopar If you could add a credential object with all the things to `.../react-learn-card/src/helpers/test.helpers.ts` I can take this bad boi across the finish line 🏁 🏎️ 💨

**Comment by gerardopar** [comment]
> @gerardopar If you could add a credential object with all the things to `.../react-learn-card/src/helpers/test.helpers.ts` I can take this bad boi across the finish line 🏁 🏎️ 💨

@smurflo2 sure thing 👍🏽

**Comment by smurflo2** [comment]
For now we're not going to do skills or notes. So hide those for now

File: packages/react-learn-card/package.json [hunk 1]
```diff
@@ -70,14 +71,13 @@
   "dependencies": {
     "@learncard/core": "workspace:*",
     "date-fns": "^2.28.0",
+    "fitty": "^2.3.6",
     "numeral": "^2.0.6",
+    "pretty-bytes": "^6.0.0",
     "qrcode.react": "^3.0.2",
-    "react": "^18.2.0",
+    "react": "link:~/WeLibraryOS/wl-react/node_modules/react",
```
Comment (goblincore) [inline packages/react-learn-card/package.json:78]: We should probably change these local links back! (same for react-dom)

File: packages/react-learn-card/package.json [hunk 2]
```diff
@@ -70,14 +71,13 @@
...
     "react-barcode": "^1.4.5",
-    "react-dom": "^18.2.0"
-  },
-  "peerDependencies": {
```
Comment (goblincore) [inline packages/react-learn-card/package.json:79]: hmm im going to do a couple of tests, but it seems like we should move react and react dom to peer dependencies and perhaps dev dependencies but remove from dependencies, will try linking this locally from learncardapp.

File: packages/react-learn-card/package.json (hunk 1 above)
Comment (smurflo2) [reply packages/react-learn-card/package.json]: Whoops, forgot about this 😅

File: packages/react-learn-card/package.json (hunk 2 above)
Comment (smurflo2) [reply packages/react-learn-card/package.json:79]: Delete react and react-dom from dependencies (they're still in peerDeps). Storybook works fine, but not really if that has other implications.. @TaylorBeeston?

Comment (goblincore) [reply packages/react-learn-card/package.json:79]: @smurflo2 Cool! I just did a test where i linked this locally in the learn-card-app repo and it worked. I think it's fine but to be safe you can add it to dev-dependencies (so react/react dom are in peerdependencies and devdepencies but not dependencies) - that seems pretty safe IMO.

**Comment by goblincore** [comment]
@smurflo2 So I tested this by importing it in to the actual learncardapp and tried it out. It works pretty well, but I was wondering if you could do a few small changes. So like this is what it looks like in learn card app:
Screenshot 2023-02-16 at 5 48 35 PM
Screenshot 2023-02-16 at 5 43 05 PM

You can see it is not respecting the font-family mouse memoirs for the titles and various other elements.
To deal with this, I usually just override it from within learn-card-app but to make this easier can you add a custom class to each these elements so I can target them easily. 

E.g. currently for example most of the elements are generic like
```
 <div className="text-center text-[#18224E] leading-[100%] text-shadow text-[32px]> <span> TITLE </span> </div>
```

which makes it hard to override. So if you just add like a unique className to it something like "vc-card-title" then I can target that and override the styles so they are correct in learn-card-app.

```
 <div className="vc-card-title-container text-center text-[#18224E] leading-[100%] text-shadow text-[32px]> <span className="vc-card-title-span"> TITLE </span> </div>

```

File: packages/react-learn-card/src/components/VCDisplayCard2/VCDisplayCard2.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        backgroundImage: credential.display?.backgroundImage
+            ? `url(${credential.display?.backgroundImage})`
+            : undefined,
+        backgroundSize: 'contain',
+        backgroundPosition: 'center',
+        backgroundAttachment: 'fixed',
+    };
+
+    return (
+        <section className="font-mouse flex flex-col items-center border-solid border-[5px] border-white h-[700px] rounded-[30px] overflow-visible z-10 max-w-[400px] relative bg-white">
```
Comment (goblincore) [inline packages/react-learn-card/src/components/VCDisplayCard2/VCDisplayCard2.tsx:113]: Could you also add a custom class to this section, i noticed you have a fixed height here set via tailwind (700px), however I will need to override this to get it to display on smaller phones

**Comment by smurflo2** [comment]
@goblincore added a bunch of class names so that whoever is consuming this can target things easier. Also added mouse font and the flip side background to main.css bc hopefully consumers don't have to manually change things in the first place

**Comment by goblincore** [comment]
Screenshot 2023-02-16 at 7 08 53 PM

@smurflo2 pushed up a fix for the above build error, TruncateTextBox needed the type def for its props updated [97be67a](https://github.com/learningeconomy/LearnCard/pull/212/commits/97be67a35bc12d65e337e13fe6b61a56e3d7c971)

**Review by goblincore — APPROVED** [review]
@smurflo2 cool this looks like it should work for me with the recent changes! thanks for working on this!

**Pull request merged**